src/util/display_layout/general.cpp

```diff
@@ -23,24 +24,57 @@
 #include <chopper/layout/input.hpp>
 
 #include <hibf/contrib/robin_hood.hpp>
+#include <hibf/contrib/std/chunk_by_view.hpp>
+#include <hibf/contrib/std/to.hpp>
 #include <hibf/sketch/hyperloglog.hpp>
 
 #include "shared.hpp"
 
-static void print_progress(size_t const percentage)
+struct progress_bar
 {
-    assert(percentage <= 100u);
-    std::cerr << '[';
-    for (size_t i{}; i < percentage; ++i)
-        std::cerr << '=';
-    if (percentage < 100u)
+    progress_bar(size_t const total) : total{total}
     {
-        std::cerr << '>';
-        for (size_t i{1u}; i < 100u - percentage; ++i)
-            std::cerr << ' ';
+        print_progress(0);
+    }
+
+    void report()
+    {
+        std::lock_guard report_guard{report_mutex};
```

Comment:
just for me since I haven't worked with lock_guards so much yet. Constructing the lock_guard here makes sure only one thread at a time can acquire a lock until the lock_guard is destructed again?